### ISSUE TITLE
Less search cost

### ISSUE THREAD

**Issue by yukang2017** [quark0/darts#12]
Hi, I run the first order cnn search with a single GPU(Titan X). It costs only 11 hours, much less than 1.5 GPU days reported in the paper. Amazing....... The log is shown bellow.
By the way, the result architectures are different in different runs. How did you pick the architecture reported in the paper?

Thank you in advance.
Yukang

07/08 05:51:51 AM train 250 1.791708e-02 99.670070 100.000000
07/08 05:53:27 AM train 300 1.792845e-02 99.667774 100.000000
07/08 05:55:03 AM train 350 1.800602e-02 99.675036 100.000000
07/08 05:56:20 AM train_acc 99.684000
07/08 05:56:20 AM valid 000 4.737676e-01 90.625000 98.437500
07/08 05:56:31 AM valid 050 4.070231e-01 89.430147 99.264706
07/08 05:56:41 AM valid 100 4.179213e-01 89.279084 99.443069
07/08 05:56:51 AM valid 150 4.126024e-01 89.093543 99.503311
07/08 05:56:53 AM valid_acc 89.110000
07/08 05:56:53 AM epoch 49 lr 1.023679e-03
07/08 05:56:53 AM genotype = Genotype(normal=[('skip_connect', 0), ('sep_conv_3x3', 1), ('skip_connect', 0), ('sep_conv_3x3', 1), ('sep_conv_3x3', 0), ('sep_conv_3x3', 1), ('skip_connect', 0), ('skip_connect', 1)], normal_concat=[2, 3, 4, 5], reduce=[('max_pool_3x3', 0), ('max_pool_3x3', 1), ('skip_connect', 2), ('max_pool_3x3', 0), ('skip_connect', 3), ('max_pool_3x3', 1), ('skip_connect', 2), ('max_pool_3x3', 0)], reduce_concat=[2, 3, 4, 5])
07/08 05:56:55 AM train 000 7.202163e-03 100.000000 100.000000
07/08 05:58:30 AM train 050 1.960013e-02 99.632353 100.000000
07/08 06:00:05 AM train 100 1.947881e-02 99.659653 100.000000
07/08 06:01:40 AM train 150 1.897141e-02 99.679222 100.000000
07/08 06:03:15 AM train 200 1.902238e-02 99.681281 100.000000
07/08 06:04:51 AM train 250 1.879712e-02 99.688745 100.000000
07/08 06:06:26 AM train 300 1.801273e-02 99.719684 100.000000
07/08 06:08:00 AM train 350 1.759578e-02 99.724003 100.000000
07/08 06:09:17 AM train_acc 99.720000
07/08 06:09:17 AM valid 000 4.404125e-01 92.187500 98.437500
07/08 06:09:28 AM valid 050 4.015100e-01 89.552696 99.356618
07/08 06:09:38 AM valid 100 4.156907e-01 89.217203 99.489480
07/08 06:09:50 AM valid 150 4.085849e-01 89.155629 99.596440
07/08 06:09:51 AM valid_acc 89.200000

**Comment by JsBlueCat** [comment]
@yukang2017  
how about your cuda version?  
I'm running this code on cuda 91 with TITAN V on ubuntu 18.04, but shit happens. 
randomly,it throws out a problem:   Python Fatal Error: GC Object already Tracked
And the core dump. 
I try to slove this problem,but failed. can you give me any suggestion?

**Comment by yukang2017** [comment]
My version is cuda 8.0 with TITAN X on ubuntu 16.04.
Could you run any other projects? Or the core dump only occurs in this project.

**Comment by JsBlueCat** [comment]
@yukang2017 
maybe it 's a problem about my cuda version.  
TITAN V can't  support cuda 8.0, 
so i run this code with pytorch=0.4.0,torchversion=0.2.1,python=3.6.6,cuda=9.1,os = ubuntu 18.04.
I  run the transfer_learning on pytorch.org, and it works.

**Comment by yukang2017** [comment]
07/07 07:24:04 PM epoch 0 lr 2.500000e-02
07/07 07:24:04 PM genotype = Genotype(normal=[('max_pool_3x3', 0), ('dil_conv_3x3', 1), ('max_pool_3x3', 1), ('dil_conv_3x3', 2), ('avg_pool_3x3', 0), ('sep_conv_3x3', 3), ('dil_conv_3x3', 4), ('avg_pool_3x3', 2)], normal_concat=[2, 3, 4, 5], reduce=[('avg_pool_3x3', 1), ('avg_pool_3x3', 0), ('sep_conv_5x5', 2), ('dil_conv_3x3', 1), ('skip_connect', 3), ('dil_conv_5x5', 0), ('sep_conv_3x3', 1), ('avg_pool_3x3', 4)], reduce_concat=[2, 3, 4, 5])
07/07 07:24:12 PM train 000 2.450172e+00 4.687500 37.500000
07/07 07:25:46 PM train 050 2.026414e+00 25.275735 79.932598
07/07 07:27:19 PM train 100 1.895834e+00 29.455446 83.524134
07/07 07:28:51 PM train 150 1.810443e+00 32.812500 85.389073
07/07 07:30:22 PM train 200 1.745026e+00 35.346704 86.901430
07/07 07:31:51 PM train 250 1.681305e+00 37.798805 88.091384
07/07 07:33:21 PM train 300 1.627401e+00 39.664659 89.124792
07/07 07:34:54 PM train 350 1.582386e+00 41.582087 89.832621
07/07 07:36:08 PM train_acc 42.896000
07/07 07:36:08 PM valid 000 1.468565e+00 59.375000 92.187500
07/07 07:36:20 PM valid 050 1.624340e+00 49.632353 91.268382
07/07 07:36:32 PM valid 100 1.627111e+00 49.566832 91.181931
07/07 07:36:42 PM valid 150 1.620696e+00 49.317053 91.721854
07/07 07:36:44 PM valid_acc 49.260000
07/07 07:36:44 PM epoch 1 lr 2.497632e-02


Hi, @quark0 this is the log of first epoch. It shows that the top1 validation accuracy achieves 48.26 after the first epoch. Is it right? Against my intuition....

**Comment by JsBlueCat** [comment]
@yukang2017  
after i set my torch to 0.3.1 and torchverion 0.2.0, i get the same result as you.
2018-07-08 20:21:03,322 gpu device = 0
2018-07-08 20:21:03,322 args = Namespace(arch_learning_rate=0.0003, arch_weight_decay=0.001, batch_size=64, cutout=False, cutout_length=16, data='../data', drop_path_prob=0.3, epochs=50, gpu=0, grad_clip=5, init_channels=16, layers=8, learning_rate=0.025, learning_rate_min=0.001, model_path='saved_models', momentum=0.9, report_freq=50, save='search-EXP-20180708-202103', seed=2, train_portion=0.5, unrolled=True, weight_decay=0.0003)
2018-07-08 20:21:09,188 param size = 1.930618MB
2018-07-08 20:21:10,322 epoch 0 lr 2.500000e-02
2018-07-08 20:21:10,323 genotype = Genotype(normal=[('max_pool_3x3', 0), ('dil_conv_3x3', 1), ('max_pool_3x3', 1), ('dil_conv_3x3', 2), ('avg_pool_3x3', 0), ('sep_conv_3x3', 3), ('dil_conv_3x3', 4), ('avg_pool_3x3', 2)], normal_concat=range(2, 6), reduce=[('avg_pool_3x3', 1), ('avg_pool_3x3', 0), ('sep_conv_5x5', 2), ('dil_conv_3x3', 1), ('skip_connect', 3), ('dil_conv_5x5', 0), ('sep_conv_3x3', 1), ('avg_pool_3x3', 4)], reduce_concat=range(2, 6))
2018-07-08 20:21:19,001 train 000 2.451357e+00 3.125000 45.312500
2018-07-08 20:23:39,291 train 050 2.030248e+00 24.540441 78.370098
2018-07-08 20:25:58,697 train 100 1.904222e+00 28.666460 82.889851
2018-07-08 20:28:18,392 train 150 1.826030e+00 31.829470 84.675083
2018-07-08 20:30:38,003 train 200 1.764723e+00 34.437189 86.287313
2018-07-08 20:32:57,431 train 250 1.707064e+00 36.522659 87.338147
2018-07-08 20:35:16,605 train 300 1.651802e+00 38.512251 88.346138
2018-07-08 20:37:36,353 train 350 1.610233e+00 40.304487 89.111467
2018-07-08 20:39:29,489 train_acc 41.864000
2018-07-08 20:39:29,698 valid 000 1.322168e+00 50.000000 95.312500
2018-07-08 20:39:34,883 valid 050 1.438617e+00 51.807598 94.117647
2018-07-08 20:39:40,098 valid 100 1.447950e+00 51.191213 94.198639
2018-07-08 20:39:45,324 valid 150 1.440904e+00 51.748758 94.153560
2018-07-08 20:39:46,229 valid_acc 51.560000
2018-07-08 20:39:46,308 epoch 1 lr 2.497632e-02
2018-07-08 20:39:46,308 genotype = Genotype(normal=[('max_pool_3x3', 0), ('max_pool_3x3', 1), ('max_pool_3x3', 0), ('max_pool_3x3', 1), ('max_pool_3x3', 0), ('sep_conv_3x3', 1), ('max_pool_3x3', 0), ('dil_conv_3x3', 3)], normal_concat=range(2, 6), reduce=[('max_pool_3x3', 0), ('max_pool_3x3', 1), ('max_pool_3x3', 0), ('max_pool_3x3', 1), ('max_pool_3x3', 0), ('dil_conv_5x5', 3), ('max_pool_3x3', 0), ('dil_conv_5x5', 4)], reduce_concat=range(2, 6))
2018-07-08 20:39:49,985 train 000 1.469631e+00 46.875000 93.750000
2018-07-08 20:42:09,535 train 050 1.228350e+00 56.250000 95.036765
2018-07-08 20:44:28,882 train 100 1.209510e+00 56.806931 94.956683

**Comment by quark0** [comment]
I don't see any problems. Fast early stage convergence should be possible for models with skip connections and a relatively small learning rate (curves for ResNets & WRNs can be found [here](https://github.com/szagoruyko/wide-residual-networks)). Also note we are optimizing both the weights and the architecture during each epoch.

**Comment by D-X-Y** [comment]
Hi @quark0 , Do other papers also run the algorithm on validation four times and report the best architecture? I just check the NAS, ENAS, PNAS, they seem not mention that.
